taichi/analysis/gather_loads_and_stores_in_blocks.cpp

```diff
@@ -0,0 +1,56 @@
+#include "taichi/ir/ir.h"
+#include <unordered_set>
+
+TLANG_NAMESPACE_BEGIN
+
+// Gather all loads, stores and atomic operations for each block.
+class LoadsAndStoresSearcher : public BasicStmtVisitor {
```

Comment:
If this is not uesd, we should save it for a future PR.

Reply:
I think this will never be used in the current code framework. Shall we just delete this file (if you like my design of `bool VariableOptimize::maybe_run;`)?